xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/saverights.vm

```diff
@@ -90,9 +97,9 @@
         #end
       #end
     #end ## end get property
-  #elseif($pers.matches("^(.*,)?${regextool.quote($fullname)}(,.*)?$")) ## The person is in a multiple selection rule, remove it from there.
-    #set($pers = $pers.replaceAll("^${fullname}(,|$)", ""))
-    #set($pers = $pers.replaceAll("(^|,)${fullname}(,|$)", "$2"))
+  #elseif($pers.matches("^(.*,)?${uorgmatch}(,.*)?$")) ## The person is in a multiple selection rule, remove it from there.
+    #set($pers = $pers.replaceAll("^${uorgmatch}(,|$)", ""))
+    #set($pers = $pers.replaceAll("(^|,)${uorgmatch}(,|$)", "$3"))
```

Comment:
Again, this feels error-prone. I find it cleaner and safer to resolve the list of user/group references and then compare. It's not like we have thousands of selected users/group, but even if it were the case, resolving a document reference is fast enough.

Reply:
Indeed, and I realized in the meantime that there was not always a third group in the match, since I introduced two regex: one for the main wiki, one for the subwikis, so this code was broken. This will be handled differently with references.